### PR TITLE
Authentication feature

### PR DESCRIPTION
**Pull request by mcolombo87**
### Summary
devise_token_auth is now implemented how authentication system for this project. 
### ScreenShots
image
### Trello
https://trello.com/c/uH3tRE0w/24-implementar-autenticaci%C3%B3n

### CODE REVIEW

File: db/migrate/20181128140935_devise_token_auth_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+class DeviseTokenAuthCreateUsers < ActiveRecord::Migration[5.2]
+  def change
+    table = :users
+    change_table(table) do |t|
+      ## Required
+      t.string :provider, :null => false, :default => "email" unless column_exists?(table, :provider)
```
Comment (facucompe) [inline db/migrate/20181128140935_devise_token_auth_create_users.rb]: Use the new syntax `null: false`

Comment (facucompe) [reply db/migrate/20181128140935_devise_token_auth_create_users.rb]: Also there is no need to use the `unless` verification, we know the column exists

File: db/migrate/20181128140935_devise_token_auth_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+      ## Recoverable
+      t.string   :reset_password_token unless column_exists?(table, :reset_password_token)
+      t.datetime :reset_password_sent_at unless column_exists?(table, :reset_password_sent_at)
+      t.boolean  :allow_password_change, :default => false unless column_exists?(table, :allow_password_change)
+
+      ## Rememberable
+      t.datetime :remember_created_at unless column_exists?(table, :created_at)
+
+      ## Trackable
```
Comment (facucompe) [inline db/migrate/20181128140935_devise_token_auth_create_users.rb]: We don't need trackable columns

File: db/migrate/20181128140935_devise_token_auth_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      t.integer  :sign_in_count, :default => 0, :null => false unless column_exists?(table, :sign_in_count)
+      t.datetime :current_sign_in_at unless column_exists?(table, :current_sign_in_at)
+      t.datetime :last_sign_in_at unless column_exists?(table, :last_sign_in_at)
+      t.string   :current_sign_in_ip unless column_exists?(table, :current_sign_in_ip)
+      t.string   :last_sign_in_ip unless column_exists?(table, :last_sign_in_ip)
+
+      ## Confirmable
```
Comment (facucompe) [inline db/migrate/20181128140935_devise_token_auth_create_users.rb]: Nor confirmable

File: db/migrate/20181128140935_devise_token_auth_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    table = :users
+    change_table(table) do |t|
+      ## Required
+      t.string :provider, :null => false, :default => "email" unless column_exists?(table, :provider)
+      t.string :uid, :null => false, :default => "" unless column_exists?(table, :uid)
+
+      ## Database authenticatable
+      t.string :encrypted_password, :null => false, :default => "" unless column_exists?(table, :encrypted_password)
+
+      ## Recoverable
```
Comment (facucompe) [inline db/migrate/20181128140935_devise_token_auth_create_users.rb:9]: We already have this columns, there is no need to add this lines in this migration

**Review by facucompe — CHANGES_REQUESTED** [review]
Please compare this PR to the `device` one

File: db/migrate/20181128140935_devise_token_auth_create_users.rb (hunk 1 above)
Comment (mcolombo87) [reply db/migrate/20181128140935_devise_token_auth_create_users.rb]: Done!.

File: db/migrate/20181128140935_devise_token_auth_create_users.rb (hunk 2 above)
Comment (mcolombo87) [reply db/migrate/20181128140935_devise_token_auth_create_users.rb]: Done!.

File: db/migrate/20181128140935_devise_token_auth_create_users.rb (hunk 3 above)
Comment (mcolombo87) [reply db/migrate/20181128140935_devise_token_auth_create_users.rb]: Done! (Again)

File: db/migrate/20181128140935_devise_token_auth_create_users.rb (hunk 4 above)
Comment (mcolombo87) [reply db/migrate/20181128140935_devise_token_auth_create_users.rb:9]: Ok

File: db/migrate/20181128140935_devise_token_auth_create_users.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+      ## Rememberable
+      # t.datetime :remember_created_at unless column_exists?(table, :created_at)
+
+      ## Lockable
+      # t.integer  :failed_attempts, :default => 0, :null => false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+      ## User Info
```
Comment (facucompe) [inline db/migrate/20181128140935_devise_token_auth_create_users.rb]: We don't need any of this fields, we already have first and last name. Also we don't want nick or image

File: db/migrate/20181128140935_devise_token_auth_create_users.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      t.string :name unless column_exists?(table, :name)
+      t.string :nickname unless column_exists?(table, :nickname)
+      t.string :image unless column_exists?(table, :image)
+
+      ## Tokens
+      t.json :tokens unless column_exists?(table, :tokens)
```
Comment (facucompe) [inline db/migrate/20181128140935_devise_token_auth_create_users.rb]: I think we did not have tokens field, remove the `unless`

File: db/migrate/20181128140935_devise_token_auth_create_users.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+      #t.timestamps
+    end
+
+    #add_index :users, :email,                unique: true unless index_exists?(:users, :email)
+    add_index :users, [:uid, :provider],     unique: true unless index_exists?(:users, [:uid, :provider])
```
Comment (facucompe) [inline db/migrate/20181128140935_devise_token_auth_create_users.rb]: Same here, remove the `unless`

**Review by facucompe — CHANGES_REQUESTED** [review]

File: db/migrate/20181128140935_devise_token_auth_create_users.rb (hunk 6 above)
Comment (mcolombo87) [reply db/migrate/20181128140935_devise_token_auth_create_users.rb]: Done!

File: db/migrate/20181128140935_devise_token_auth_create_users.rb (hunk 7 above)
Comment (mcolombo87) [reply db/migrate/20181128140935_devise_token_auth_create_users.rb]: Done!

**Review by facucompe — APPROVED** [review]

**Pull request merged**